subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -344,8 +344,10 @@ int lll_adv_scan_req_report(struct lll_adv *lll, struct pdu_adv *pdu_adv_rx,
 	pdu_len = offsetof(struct pdu_adv, payload) + pdu_adv_rx->len;
 	memcpy(pdu_adv, pdu_adv_rx, pdu_len);
 
-	node_rx->hdr.rx_ftr.rssi = (rssi_ready) ? (radio_rssi_get() & 0x7f) :
-						  0x7f;
+	node_rx->hdr.rx_ftr.rssi = (rssi_ready) ?
+				   (radio_rssi_get() &
+				    BT_HCI_LE_ADV_TX_POWER_NO_PREF) :
+				   BT_HCI_LE_ADV_TX_POWER_NO_PREF;
```

Comment:
While the value of this is correct, I would prefer (for readability) a separate #define for the rssi.